protocol/encoding.go

```diff
@@ -0,0 +1,147 @@
+// Defines methods/functions to encode/decode messages between client and server.
+// Currently this module supports JSON marshal/unmarshal only.
+// Protobuf would be supported in the feature.
+
+package protocol
+
+import (
+	"encoding/base64"
+	"encoding/json"
+
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+var b64en = base64.StdEncoding.EncodeToString
+
+func MarshalTemporaryBinding(tb *merkletree.TemporaryBinding) ([]byte, error) {
+	if tb == nil {
+		return nil, nil
+	}
+	return json.Marshal(tb)
+}
+
+func MarshalSTR(str *merkletree.SignedTreeRoot) ([]byte, error) {
+	policies, err := json.Marshal(str.Policies)
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(&struct {
```

Comment:
I'm going to backtrack a little bit.  Maybe I don't know what I'm talking about.

It feels like the encoding of the messages should be part of the server, since that's where the decision about JSON/Protobuf/etc. will happen.  But the message struct should be part of the protocol, and they should define the interface with the directory.

Help me think this through.